Installed Collected CLI doc formatting fixes (#4376)
SumoLogic/sumologic-documentation: docs/send-data/installed-collectors/collector-installation-reference/parameters-command-line-installer.md

---
id: parameters-command-line-installer
title: Parameters for the Command Line Installer
description: This topic lists the parameters that are available when installing a Sumo Logic collector using the command line installer.
---

This page provides the available parameters for Collector installation using the command line installer.

:::important
Collector versions 19.253-26+ support wrapper configuration parameters.
:::

> For details and example commands on Collector installation, see [Install a Collector on Linux](../linux.md), [Install a Collector on MacOS](../macos.md), and [Install a Collector on Windows](../windows.md).

Reviewer: ```suggestion
For details and example commands on Collector installation, see [Install a Collector on Linux](../linux.md), [Install a Collector on macOS](../macos.md), and [Install a Collector on Windows](../windows.md).
```

## Unique parameters

Here are the parameters unique to the command line installer:

---
### `-console`

Only has an effect when used with `-q`. Causes the installer to send progress messages to the console (whether you are running the installer in UI or command-line mode.)

On Windows, for this option to take effect, you must run the installer with start `/wait`. For example:

`start /wait installer.exe -q -console`

---
### `-dir [directory]`

Sets a different installation directory than the default.

---
### `-q`

> Causes the installer to run silently, which means you won't be prompted to supply installation parameters. For any installation parameter that you do not define at the command line, Sumo will use a default value. No output is sent to the console during installation, unless you also use the `-console` parameter.

Reviewer: ```suggestion
Causes the installer to run silently, which means you won't be prompted to supply installation parameters. For any installation parameter that you do not define at the command line, Sumo Logic will use a default value. No output is sent to the console during installation, unless you also use the `-console` parameter.
```

Note that for a new Collector installation, even during a silent installation, you will be prompted to provide an authorization token, or an Access ID and Access Key.

---
### `-varfile [fileName]`

Instead of passing arguments via the command line, you can specify a file that contains the arguments (without the `-V` flag). Since many operating systems allow a user to list information about processes that are owned by other users, we recommend using the `-varfile` option to specify Sumo Logic credentials.

Example: To pass credentials when installing the Collector, you can create a file called **sumo_credentials.txt** with the following contents:

`sumo.accessid=<access id>
sumo.accesskey=<access key>
proxy.host=<proxy host>
proxy.port=<proxy port>`

and pass this file via the command line:

`sudo ./SumoCollector.sh -q -varfile /path/to/sumo_credentials.txt`

Parameters for `-q` and `-dir` must be passed in the command line and cannot be used within a varfile.

---
### `-Vcollector.secureFiles=[true|false]`

> By default, new Collector installations will use [Enhanced File System Security](enhanced-file-system-security-installed-collectors.md). To disable this feature during installation, use the command line argument: `-Vcollector.secureFiles=false`

Reviewer: ```suggestion
By default, new Collector installations will use [enhanced file system security](enhanced-file-system-security-installed-collectors.md). To disable this feature during installation, use the command line argument: `-Vcollector.secureFiles=false`
```

---
### `-VrunAs.username=[username]`

When set, the Collector will run as the specified user (Windows and Linux). For Windows, the user account needs `Log on as a Service` permission.

---
### `-VwinRunAs.password=[password]`

(Windows only) When set in conjunction with `-VrunAs.username`, the Collector will run as the specified user with the specified password.

Many operating systems allow a user to list information about processes that are owned by other users. This information could include command line arguments. For enhanced security, we recommend using the `-varfile` option to specify Windows RunAs password.

## Authentication parameters

> The following table shows parameters that relate to authenticating the Collector.

Reviewer: ```suggestion
Following are parameters that relate to authenticating the Collector.
```
